src/azure-cli/azure/cli/command_modules/profile/_validators.py

```diff
@@ -0,0 +1,37 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import uuid
+
+from azure.cli.core.util import should_disable_connection_verify
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+def _is_guid(guid):
+    try:
+        uuid.UUID(guid)
+        return True
+    except ValueError:
+        return False
+
+
+def validate_tenant(cmd, namespace):
+    """
+    Make sure tenant is a GUID. If domain name is provided, resolve to GUID.
+    https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc#fetch-the-openid-connect-metadata-document
+    """
+    if not _is_guid(namespace.tenant):
+        import requests
+        active_directory_endpoint = cmd.cli_ctx.cloud.endpoints.active_directory
+        url = '{}/{}/.well-known/openid-configuration'.format(active_directory_endpoint, namespace.tenant)
```

Comment:
`v1.0` endpoint is now deprecated, so we need to migrate to `v2.0`:

https://login.microsoftonline.com/azuresdkteam.onmicrosoft.com/.well-known/openid-configuration ->
https://login.microsoftonline.com/azuresdkteam.onmicrosoft.com/v2.0/.well-known/openid-configuration

Official documentation on `/.well-known/openid-configuration`: https://learn.microsoft.com/en-us/entra/identity-platform/v2-protocols-oidc#find-your-apps-openid-configuration-document-uri